src/cluster_migrateslots.c

```diff
@@ -823,6 +844,9 @@ bool clusterSlotFailoverGranted(int slot) {
  * source will attempt to migrate the slot ranges to the specified target
  * node. */
 void clusterCommandMigrateSlots(client *c) {
+    if (moduleVerifyAllModulesAllowAtomicSlotMigrationOrReply(c) == C_ERR) {
```

Comment:
We should also do this check on the target if for some reason the cluster is non-homogenous in the modules that are running.